src/coreclr/gc/gc.cpp

```diff
@@ -3850,6 +3850,39 @@ void region_allocator::delete_region (uint8_t* region_start)
 
     leave_spin_lock();
 }
+
+bool region_allocator::should_delete_region (uint8_t* start, int num_free_regions, int num_free_large_regions)
+{
+    size_t address_space_available = global_region_end - global_region_start;
+#ifdef MULTIPLE_HEAPS
+    size_t address_space_available_per_heap = address_space_available / gc_heap::n_heaps;
+#else //MULTIPLE_HEAPS
+    size_t address_space_available_per_heap = address_space_available;
+#endif //MULTIPLE_HEAPS
+
+    size_t address_space_in_free_regions = (num_free_regions       * region_alignment      ) +
+                                           (num_free_large_regions * large_region_alignment);
+
+    // don't keep more than a quarter of the address space in the per-heap free lists
```

Comment:
is the "no more than a quarter" heuristic similar for segments?

Reply:
No, there is no equivalent for segments, I think. This heuristic is meant to prevent "hoarding" of regions on the per-heap freelists.

Reply:
ok I see. so assume we would need to tune based on various perf scenarios?

Reply:
it might be better to use the ratio of (what's in free_regions/free_large_regions / heap size) instead of (what's in free_regions/free_large_regions / max reserved for that heap) - if we had the conserve memory checked in we could use that ratio; lack of that we could use 65%, taking from dt_high_frag_p.